### PR TITLE
Add Total_Build time to end of build_times.txt

### PR DESCRIPTION
**Pull request by jgfouca**
... in e3sm's provenance code.

Also, minor fix for list_e3sm_tests.

Test suite: by hand
Test baseline:
Test namelist changes:
Test status: [bit for bit, roundoff, climate changing]

User interface changes?:

Update gh-pages html (Y/N)?:

### CODE REVIEW

**Review by jedwards4b — APPROVED** [review]

**Comment by jedwards4b** [comment]
@jgfouca these should be fixed by a merge of master.

**Comment by jgfouca** [comment]
@jedwards4b , git is telling me my branch is up to date.

**Comment by jgfouca** [comment]
Ah, ok. Thanks

**Comment by sarats** [comment]
@jgfouca It would be great if you can you apply this change to maint-2.0 as well.

File: CIME/provenance.py [hunk 1]
```diff
@@ -54,8 +55,16 @@ def _extract_times(zipfiles, target_file):
                     target, the_time = items[1], items[-2]
                     contents += "{} {}\n".format(target, the_time)
 
+        stat, output, _ = run_cmd("zgrep -E '^user [0-9.]+$' {}".format(zipfile))
```
Comment (sarats) [inline CIME/provenance.py:58]: I think we should use the "real" time reported by the time command to account for all the I/O and system/kernel overheads involved.

File: CIME/provenance.py [hunk 2]
```diff
@@ -44,6 +44,7 @@ def _get_batch_job_id_for_syslog(case):
 def _extract_times(zipfiles, target_file):
 
     contents = "Target Build_time\n"
+    total_build_time = 0.0
     for zipfile in zipfiles:
         stat, output, _ = run_cmd("zgrep 'built in' {}".format(zipfile))
```
Comment (sarats) [inline CIME/provenance.py:49]: In a similar vein, is the target build time reporting elapsed (real) or user time?

Comment (jgfouca) [reply CIME/provenance.py:49]: The individual targets use the python `time` library, so should not have the issue of distinguishing between user and real times.

File: CIME/provenance.py (hunk 1 above)
Comment (jgfouca) [reply CIME/provenance.py:58]: I think you're right.

File: CIME/provenance.py (hunk 2 above)
Comment (sarats) [reply CIME/provenance.py:49]: Just for my reference, where is the CMAKE wrapper you use for recording this time info?

Comment (jgfouca) [reply CIME/provenance.py:49]: @sarats, `cime/CIME/Tools/e3sm_compile_wrap.py` . This is the customizable target wrapper for E3SM. The default version (version the lives in the repo) just does timings.

**Pull request merged**